csharp/ql/src/semmle/code/csharp/Unification.qll

```diff
@@ -10,9 +10,69 @@ private import Caching
  * equal modulo identity conversions and type parameters.
  */
 module Gvn {
+  /** Gets the qualified name of type `t`. */
+  string getQualifiedName(Type t) {
+    if not t instanceof NestedType or t.(NestedType).getDeclaringType() instanceof GenericType
+    then result = t.getName()
+    else result = getQualifiedName(t.(NestedType).getDeclaringType()) + "." + t.getName()
+  }
+
+  /**
+   * A generic type. This is either a type with a type parameter, a type with
+   * a type argument, or a nested type with a generic enclosing type.
+   */
+  class GenericType extends Type {
```

Comment:
The reason I went with `GenericType` is because it covers both parameterized types and constructed types. We also use that name [elsewhere](https://github.com/github/codeql/blob/master/csharp/ql/src/semmle/code/csharp/Conversion.qll#L101).

Reply:
The difference is that it also includes nested types that are only indirectly generic. Well, I don't want to quibble too much over a name, it was just a suggestion.